### DISCUSSION TITLE
What does `permanent` actually do for redirects?

### DISCUSSION THREAD

**Discussion by aminick** [vercel/next.js#29556] [Help] [answered]
From the [docs](https://nextjs.org/docs/api-reference/next.config.js/redirects):
![image](https://user-images.githubusercontent.com/5182324/135702941-e1f35f7b-f5d3-4d53-a8f2-3577fc807c8d.png)

I am wondering what does `permanent` actually do?

`permanent if the redirect is permanent or not.` This was not really helpful from the official doc.

**Comment by ijjk** [comment, marked as answer]
Hi, permanent redirects use the [`308` status code](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/308) which tells browsers/search engines they can cache the redirect forever and it will not change. You need to be careful with this type of redirect because it can't be changed in most cases after it has been cached by a client.

**Reply by aminick** [reply]
Thanks, I suggest this should be added to the Doc, since it has a big effect for the website behavior.

**Reply by aminick** [reply]
@ijjk Just want to confirm with you, even if the website is re-deployed with `permanent` set to `false`, it is likely to be useless if the user already has that previous cache?

**Reply by ijjk** [reply]
Yeah if the deployment is using the same domain it will most likely still be cached unless the client manually busts the cache (e.g. a hard refresh cmd + shift + r in chrome on mac irc). I opened a PR here tweaking this note in the documentation https://github.com/vercel/next.js/discussions/29556

**Reply by aminick** [reply]
Thanks
